### PR TITLE
Data cleaning region tags

### PR DESCRIPTION
**Pull request by leahecole**
## Description
Adds region tags for the data cleaning section, removes the gender column from the dataframe. Link to doc in staging available if you ping me :) 

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Comment by leahecole** [comment]
fixing lint now ugh 🙈

**Review by busunkim96 — APPROVED** [review]

File: data-science-onramp/data-cleaning/clean.py [hunk 1]
```diff
@@ -129,26 +137,33 @@ def compute_end_udf(duration, start, end):
...
     spark = SparkSession.builder.appName("data_cleaning").getOrCreate()
 
-    # Load data into dataframe if table exists
+    # Load data into temporary dataframe if table exists, drop unused column and load into new dataframe
+    # (using a temporary dataframe is an alternative to calling df.drop(column_name).collect() and is used
+    # to avoid OOM issues)
     try:
-        df = spark.read.format("bigquery").option("table", TABLE).load()
+        temp_df = spark.read.format("bigquery").option("table", TABLE).load()
+        df = temp_df.drop("gender")
```
Comment (bradmiro) [inline data-science-onramp/data-cleaning/clean.py:155]: Should this be more explicitly explained in the code why we are removing gender? Either way, I would move it outside of this try-block. It's also ok to do `df = df.<operation>` and not use the `temp_df` var name as this ends up happening in multiple other places in the script.

File: data-science-onramp/data-cleaning/clean_test.py [hunk 2]
```diff
@@ -109,7 +106,7 @@ def setup_and_teardown_cluster():
         project_id=PROJECT_ID,
         region=CLUSTER_REGION,
         cluster_name=DATAPROC_CLUSTER,
-        timeout=300
+        timeout=300,
```
Comment (bradmiro) [inline data-science-onramp/data-cleaning/clean_test.py:109]: This doesn't hurt but you shouldn't need to do this. Were there timeout errors deleting the cluster?

**Review by bradmiro — CHANGES_REQUESTED** [review]

File: data-science-onramp/data-cleaning/clean.py (hunk 1 above)
Comment (leahecole) [reply data-science-onramp/data-cleaning/clean.py:155]: Np for moving it outside of try block and making that df = df.operation change. Reasoning for removal of the column is in the **Putting it together** section of the doc I sent you - do you think it also needs to be mentioned here?

File: data-science-onramp/data-cleaning/clean_test.py (hunk 2 above)
Comment (leahecole) [reply data-science-onramp/data-cleaning/clean_test.py:109]: leftover from Tushar - can remove if need be

**Comment by leahecole** [comment]
ugh, ofc these were passing locally. I'll figure out why they don't pass here!

File: data-science-onramp/data-cleaning/clean.py (hunk 1 above)
Comment (bradmiro) [reply data-science-onramp/data-cleaning/clean.py:155]: I think that's fair, maybe it should have its own region tag then?

File: data-science-onramp/data-cleaning/clean_test.py (hunk 2 above)
Comment (bradmiro) [reply data-science-onramp/data-cleaning/clean_test.py:109]: Removing is fine unless it somehow breaks

File: data-science-onramp/data-cleaning/clean.py (hunk 1 above)
Comment (leahecole) [reply data-science-onramp/data-cleaning/clean.py]: Added region tag, but have to wait a few min before re-staging

File: data-science-onramp/data-cleaning/clean_test.py (hunk 2 above)
Comment (leahecole) [reply data-science-onramp/data-cleaning/clean_test.py:109]: Will remove next, just testing locally now

Comment (leahecole) [reply data-science-onramp/data-cleaning/clean_test.py]: local tests passed, just pushed removal

File: data-science-onramp/data-cleaning/clean.py [hunk 3]
```diff
@@ -147,17 +147,21 @@ def compute_end_udf(duration, start, end):
...
     except Py4JJavaError as e:
         raise Exception(f"Error reading {TABLE}") from e
 
-    # [END datascienceonramp_sparksession]
-    # [START datascienceonramp_sparksingleudfs]
+# [END datascienceonramp_sparksession]
+
+# [START datascienceonramp_removecolumn]
+    # remove unused column
+    # not using .collect() notation to avoid OOM
```
Comment (bradmiro) [inline data-science-onramp/data-cleaning/clean.py:160]: No need to mention `.collect()` here.

Comment (leahecole) [reply data-science-onramp/data-cleaning/clean.py:160]: done

**Review by bradmiro — APPROVED** [review]

**Pull request merged**